### DISCUSSION TITLE
How can I get syntax highlighting with `toggle diff`

### DISCUSSION THREAD

**Discussion by toshmukhamedov** [NeogitOrg/neogit#1187] [Q&A] [answered]
<img width="1438" alt="image" src="https://github.com/NeogitOrg/neogit/assets/79999469/1dfe5646-c937-4713-b2d4-02fc7e4d0b60">

**Comment by CKolkey** [comment, marked as answer]
Not possible, currently. It might be interesting to play with using treesitter injections for hunks, though, that might work.

**Reply by toshmukhamedov** [reply]
any plans to do this?

**Reply by CKolkey** [reply]
I'm rewriting the status buffer right now - when thats done this would be a bit easier. It'd be cool, yeah

**Reply by radloffl** [reply]
This ever become possible? Hoping to get syntax coloring in the status buffer.

**Reply by CKolkey** [reply]
No, not yet. There's a PR that kinda adds this via an external tool, but I have had zero time to take a look at it. Babies eat your free time, turns out.

**Reply by radloffl** [reply]
Oh very cool, didn't see that PR (#1593 for future lurkers). Congrats on the kiddo! I'm very new to Neovim so might not be much help on the technical side right now but if there is any grunt work or docs needed that can help out with that one let me know!

**Edit** kiddos!

**Reply by danrasmuson** [reply]
I do think this is possible with treesitter injections for hunks. It probably will not be the easiest code to get working. I've achieved using treesitter to get `nushell` highlighting in my lua code whenever it sees `[[#nu` in a multiline lua string. It works great once its setup. In this scenario we would not have the luxury of having the language marker in the code so it would need to be applied programmatically in some other way. Including this screenshot in case it proves as a helpful general reference

![CleanShot 2025-02-23 at 15 04 24@2x](https://github.com/user-attachments/assets/15a691f8-99e4-4536-a9b1-11eecf4ec399)

**Reply by CKolkey** [reply]
I tried implementing something like that a while back, but ran into some issues:
- defining a range of lines for the injected region wasn't doable via api, but more importantly...
- diff's may not be syntactically valid, missing content before or after the hunk, which would render the tree unusable. So you'd have to do something like parsing the whole file, reading the highlights from that tree, and apply them via extmarks... or something like that. Never really explored that though.


Anyways, it's still on my todo list, but probably not for some time.

**Reply by danrasmuson** [reply]
Yea. Good point. I hadn't thought of that.
